master/internal/stream/test_util.go

```diff
@@ -24,7 +24,7 @@ const (
 // otherwise, returns the MarshallableMsg that the streamer sends.
 func testPrepareFunc(i stream.MarshallableMsg) interface{} {
 	switch msg := i.(type) {
-	case stream.UpsertMsg:
+	case *stream.UpsertMsg:
```

Comment:
curious, why do we need to change to pointer types.

Reply:
`testPrepareFunc` is called in `Broadcast()` https://github.com/determined-ai/determined/pull/9164/files#diff-ab55b57cea1409ab6f97b1352891adb036d83ee04a82f8428eee733d70351cf6R247:
```
msg = sub.Streamer.PrepareFn(recordCache.UpsertMsg)
```
The type of `recordCache.UpsertMsg` is `*stream.UpsertMsg`.